landlab/components/stream_power/sed_flux_dep_incision.py

```diff
@@ -646,46 +669,33 @@ def sed_flux_fn_gen(rel_sed_flux_in):
         sed_flux_out = rel_sed_flux * trans_cap_vol_out
         return dz, sed_flux_out, rel_sed_flux, error_in_sed_flux_fn
 
-    def erode(self, dt, flooded_depths=None, **kwds):
-        """Erode and deposit on the channel bed for a duration of *dt*.
+    def run_one_step(self, dt):
+        """Run the component across one timestep increment, dt.
 
         Erosion occurs according to the sediment dependent rules specified
-        during initialization.
+        during initialization. Method is fully equivalent to the :func:`erode`
+        method.
 
         Parameters
         ----------
         dt : float (years, only!)
             Timestep for which to run the component.
-        flooded_depths : array or field name (m)
-            Depths of flooding at each node, zero where no lake. Note that the
-            component will dynamically update this array as it fills nodes
-            with sediment (...but does NOT update any other related lake
-            fields).
         """
-        if self._grid.at_node["flow__receiver_node"].size != self._grid.size("node"):
-            msg = (
-                "A route-to-multiple flow director has been "
-                "run on this grid. The landlab development team has not "
-                "verified that SedDepEroder is compatible with "
-                "route-to-multiple methods. Please open a GitHub Issue "
-                "to start this process."
-            )
-            raise NotImplementedError(msg)
 
-        grid = self.grid
+        grid = self._grid
         node_z = grid.at_node["topographic__elevation"]
         node_A = grid.at_node["drainage_area"]
         flow_receiver = grid.at_node["flow__receiver_node"]
         s_in = grid.at_node["flow__upstream_node_order"]
         node_S = grid.at_node["topographic__steepest_slope"]
 
-        if type(flooded_depths) is str:
-            flooded_depths = grid.at_node[flooded_depths]
+        if type(self._flooded_depths) is str:
```

Comment:
These should all be `if isinstance(...)`. I notice now that the `if type(...)` pattern is fairly widely used throughout landlab. How about we fix this in another pull request?

Reply:
Fixing all the `type()` to `isinstance()` was a good bus task. 🚌 ! done (well at least within components)